tools/tungsten/core/src/com/google/android/filament/tungsten/model/Nodes.kt

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2018 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.filament.tungsten.model
+
+import com.google.android.filament.tungsten.compiler.GraphCompiler
+import com.google.android.filament.tungsten.compiler.Expression
+
+private val adderNodeCompile = fun(node: Node, compiler: GraphCompiler): Node {
+    val outputSlot = node.getOutputSlot("result")
+
+    val a = compiler.compileAndRetrieveVariable(node.getInputSlot("a"))
+    val b = compiler.compileAndRetrieveVariable(node.getInputSlot("b"))
+
+    val aExpression = a?.symbol ?: "float3(0.0f, 0.0f, 0.0f)"
```

Comment:
You can just write float3(0.0). You need to drop the 'f' prefix btw, it's only valid in desktop GL, not GLES